master/internal/experiment.go

```diff
@@ -139,6 +140,7 @@ func newExperiment(
 		m.system,
 		poolName,
 		resources.SlotsPerTrial(),
+		restored,
```

Comment:
To me it might make more sense to skip the restore here and avoid passing this restore check logic all the way to the resource manager.

Reply:
Do you mean skip calling `ValidateResourcePoolAvailability` if the experiment is restored?

Reply:
It may be that command should be updated also, as an admin if the server is reconfigured with a lower limit and then any associated processes stopped it may be strange to see them recreated without regard to the new limit.

Reply:
the server `config.Provider.MaxInstances` variable which is used in `ValidateResourcePoolAvailability` along with the instance type determines the limit (for AWS & GCP). if either of those is reconfigured a restored experiment might be out of those desired soft limits. for the static agents we current validate on start and the purpose of this PR is to not validate on restart, altho perhaps we should not validate static agent resources.

Reply:
Yeah that makes sense but then I think we need to make commands have consistent behavior

Reply:
it does look a little ugly and feels like static agents are somewhat of a special case, just want to make sure we do the correct thing. Any thoughts on the best course of action @ioga & @eecsliu?

Reply:
looking again at the command code, I don't think we ever restore commands.

Reply:
I'm a little confused because it looks like `restored` is only checked if the provider is nil, so we'd never really run into this case where we need the special restore arg when the max instances have changed?

Otherwise, based on the changes, I'd also support just skipping the validation. Changing your static configuration while there are jobs running sounds to me like the admin should make sure things are pausing or killed

Reply:
correct, `restored` is meant to impact the static agent configuration rather than the AWS or GCP provider scenario (because that is where the issue occurs). my point was that it would be a good idea to avoid changing the logic for the other scenarios (eg skipping the validation), due to configuration changes and the need to then explicitly kill experiments. it would also create some interesting cluster statistics as we exceed capacity.

Reply:
>looking again at the command code, I don't think we ever restore commands.

we have https://github.com/determined-ai/determined/blob/main/master/internal/command/command.go#L190 which restores things that have already been running, but it doesn't schedule new allocations which is the problem here.

I like the idea of skipping `ValidateResourcePoolAvailability` for the restored experiments. Right now we have these calls on RM interface:
- ValidateResources
- ValidateResourcePool
- ValidateResourcePoolAvailability

which do different checks depending on the resource manager, e.g. ValidateResources [does nothing on k8s](https://github.com/determined-ai/determined/blob/main/master/internal/rm/kubernetesrm/kubernetes_resource_manager.go#L123) even though there's [the underlying code](https://github.com/determined-ai/determined/blob/main/master/internal/rm/kubernetesrm/kubernetes_resource_manager.go#L327) which is supposed to make it do the same as agent rm. I think this needs to be fixed, so that even when we skip `ValidateResourcePoolAvailability`, the preceding `ValidateResources` or `ValidateResourcePool` calls would at least check if the requested resource pool exists.

Reply:
Gotcha, thanks everyone for the feedback! I've created DET-9579 to follow up with the k8s `ValidateResource` methods and simplified the code to skip on restore.